model-view-viewmodel/README.md

```diff
@@ -0,0 +1,227 @@
+---
+layout: pattern
+title: Model-View-ViewModel
+folder: model-view-viewmodel
+permalink: /patterns/model-view-viewmodel/
+categories: Architectural
+tags:
+ - Decoupling
+---
+
+## Also known as
+
+model–view–binder
+
+## Intent
+
+To apply "Separation of Concerns" to separate the logic from the UI components and allow developers to work on UI without affecting the logic and vice versa.
+
+## Explanation
+
+Wikipedia says
+
+> Model–view–viewmodel (MVVM) is a software architectural pattern that facilitates the separation of the development of the graphical user interface (the view) – be it via a markup language or GUI code – from the development of the business logic or back-end logic (the model) so that the view is not dependent on any specific model platform. 
+
+**Programmatic Example**
+
+Android Implementation:
```

Comment:
It might be too much to have both Android and zkoss implementation here. I would suggest picking only one of them.

Reply:
done